doc/ondemand_design.md

```diff
@@ -632,7 +632,11 @@ We iterate through object instances using `field` instances which represent key-
 is accessible by the `value()` method whereas the key is accessible by the `key()` method.
 The keys are treated differently than values are made available as as special type `raw_json_string`
 which is a lightweight type that is meant to be used on a temporary basis, amost solely for
-direct raw ASCII comparisons (`field.key() == "mykey"`). If you occasionally need to access and store the
+direct raw ASCII comparisons: `key().raw()` provides direct access to the unescaped string.
+You can compare `key()` with unescaped C strings (e.g., `key()=="test"`). Importantly,
+the C string must not contain an unescaped quote character (`"`) which you can check with
+`raw_json_string::is_free_from_unescaped_quote("test")`.
```

Comment:
I think the more common case is covered by "you can compare directly with a raw string as long as the string you are comparing it to does not have escape characters in it." The unescaped quote character in the raw JSON is not a factor there.

Reply:
In fact, I don't think this restriction ever applies, actually. The only way we could screw it up is if the user compares something with a raw string terminated with a single backslash, which is invalid JSON anyway. I think the best way to state the restriction is "if you want to compare with the raw JSON key, you must compare against a key with valid JSON (which requires escape characters if they will contain control characters, newlines, tabs, backslash and quote)."

The restriction we need to worry about is \uXXXX, actually. That's the best way to accidentally get a false negative. "This can return a false negative if the JSON file contains keys with \uXXXX in it. If this is a concern, you can call safe_equal() or is_free_from_unicode_escapes()."

Reply:
Oh! You're talking about pre-validating your own keys. That makes sense. I think the right check (as noted in another comment) is is_normalized_json_string(), ensuring that one-character escapes are used, \u is only used for otherwise unrepresentable characters, and everything else is raw UTF-8.

However, what you did is a big step better than what we have already, which I put together rather haphazardly.

Reply:
@jkeiser Suppose that you have the following json `{"a":1}` and I do `key() == "a\":1}bababababbababa....."`. The comparison could possibly lead to a buffer overrun... can't it? Clearly, `a":1}bababababbababa.....` is not a valid JSON string, but imagine that I am an adversary trying to make simdjson crash.

Reply:
Yep! I wasn't thinking about keys being provided by users--this is more designed around keys provided by the developer. Totally makes sense.